### PR TITLE
Allows the AI and cyborgs to join midround

### PR DESCRIPTION
**Pull request by Cenrus**
## About this pull request

The code for midround AI connections was functional, but no midround AI or cyborg positions were available. Now it is possible for the AI and one cyborg to join midround. The midround join screen has been updated to suit these changes

new

## Why it's good for the game

More job options for players who join the shift late.

## Changelog

:cl:
add: AIs and cyborgs can now latejoin.
/:cl:

### CODE REVIEW

**Comment by qwertyquerty** [comment]
TG mirror?

File: code/modules/mob/dead/new_player/new_player.dm [hunk 1]
```diff
@@ -49,7 +49,7 @@
 	if(!IsGuestKey(src.key))
 		if (SSdbcore.Connect())
 			var/isadmin = 0
-			if(src.client?.holder)
+			if(src.client && src.client.holder)
```
Comment (qwertyquerty) [inline code/modules/mob/dead/new_player/new_player.dm]: Change this line back to how it was.

File: code/modules/mob/dead/new_player/new_player.dm [hunk 2]
```diff
@@ -138,9 +138,6 @@
 		ViewManifest()
 
 	if(href_list["SelectedJob"])
-		if(!SSticker || !SSticker.IsRoundInProgress())
```
Comment (qwertyquerty) [inline code/modules/mob/dead/new_player/new_player.dm:141]: Don't remove this

**Comment by Cenrus** [comment]
Mirrored this https://github.com/tgstation/tgstation/pull/44193

**Comment by Cenrus** [comment]
the check passed before these changes

**Comment by MarkSuckerberg** [comment]
I don't think you need to change the actual code, just the job slot amount. You can admin set it to have a job slot for AIs and it works.

File: config/jobs.txt [hunk 3]
```diff
@@ -42,5 +42,5 @@ Warden=1,1
 Detective=1,1
 Security Officer=5,5
 
-AI=0,1
-Cyborg=0,1
+AI=1,1
+Cyborg=1,1
```
Comment (MarkSuckerberg) [inline config/jobs.txt:46]: These lines are all that need to be changed, nothing else.

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

**Comment by Cenrus** [comment]
I'll do the improved midjoin screen in another PR

**Pull request closed without merging**